### PR TITLE
Issue-129: Edit Program

### LINKED ISSUE #129: Edit Existing Programs
Allow for editing existing programs in the program dialog.

### PR DESCRIPTION
**Pull request by mckoon**
* Adds create and edit functions to programFormDialogService.
* Adds trackBy to organizations ng-repeat to fix object equality.
* Opens program-form-dialog from Edit button on program-details.

Closes #129

### CODE REVIEW

**Comment by mckoon** [comment]
# Screenshot:

screen shot 2016-10-17 at 8 39 35 pm

**Comment by mckoon** [comment]
Running at https://hmhb-mckoon.herokuapp.com/views/program-details/4

File: client/src/main/webjar/program-details/program-details.component.js [hunk 1]
```diff
@@ -6,32 +6,57 @@
...
          * Handles success loading the MapQuery for the program.
          * @param mapQuery response.
          * @returns mapQuery input or rejected promise on error.
          */
         function handleMapQuerySearchSuccess(mapQuery) {
 
             if (mapQuery.result && angular.isArray(mapQuery.result.programs) && mapQuery.result.programs.length === 1) {
-                programDetails.program = mapQuery.result.programs[0];
+                programDetails.program = new Program(
+                    mapQuery.result.programs[0]
```
Comment (RyanBard) [inline client/src/main/webjar/program-details/program-details.component.js]: Maybe it doesn't matter b/c the object is flat, so the prototypical inheritance is enough.

Comment (mckoon) [inline client/src/main/webjar/program-details/program-details.component.js]: Yeah, I probably should copy it.

**Comment by mckoon** [comment]
While testing this last night, I forgot that we now save ProgramAreas, and I was equating them being blank with not saving yet.

I remember this morning that is not the case.  I need to loop over the program areas and set the selected status when loading the edit dialog.

**Comment by RyanBard** [comment]
Is it problematic for you that ProgramAreas is implemented as a Set<ProgramArea> instead of a Map<Long, ProgramArea>?  I can look into that if it's annoying to the client.

**Comment by RyanBard** [comment]
ditto for Set<County>

**Pull request merged**